ygen/enumgen.go

```diff
@@ -363,6 +422,61 @@ func findEnumSet(entries map[string]*yang.Entry, compressPaths, noUnderscores, s
 	// where there are erroneous config/state differences).
 	sort.Strings(enumPaths)
 
+	s := newEnumGenState()
+
+	// This is the first of two passes over the input enum entries. The
+	// purpose of this pass is to resolve the names of all the enumerated
```

Comment:
`The purpose of this pass is to establish what the default name of each of the enumerated types that are to be output in the code is. At this stage, we just try and calculate the complete set, rather than resolving any clashes in names that might occur.`